File: Sumroot.java
// Time Complexity :O(n)
// Space Complexity :O(n)
// Did this code successfully run on Leetcode :Yes
// Any problem you faced while coding this :


// Your code here along with comments explaining your approach
//we kept global sum to calculate sum at each node and we are passing the same sum to the right and left tree 


 //Definition for a binary tree node.
 public class TreeNode {
      int val;
      TreeNode left;
      TreeNode right;
      TreeNode() {}
      TreeNode(int val) { this.val = val; }
      TreeNode(int val, TreeNode left, TreeNode right) {
         this.val = val;
         this.left = left;
         this.right = right;
      }
 }
 
class Solution {
    int output = 0;
    public int sumNumbers(TreeNode root) {
        if(root==null){
            return output;
        }
        recurrsive(root,0);
        
        return output;
        
    }
    private void recurrsive(TreeNode root,int prev){
        if(root==null){
            return;
        }
        
        int curr = prev*10+root.val;
        
        if(root.left==null&&root.right==null){
            output+=curr;
        }  
        
        recurrsive(root.left,curr);
        recurrsive(root.right,curr);
    }
}

File: SumRoot.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    int result;
    public int SumNumbers(TreeNode root) {
        result = 0;
        helper(root, 0);
        return result;
    }
    private void helper(TreeNode root, int currSum)
    {
        if(root == null) return;
        currSum = currSum * 10 + root.val;
        helper(root.left, currSum);
        if(root.left == null && root.right == null)
        {
            result += currSum;
        }
        helper(root.right, currSum);

    }
}
